src/NativeScript/ObjC/ObjCPrimitiveTypes.mm

```diff
@@ -23,13 +24,23 @@ static JSValue objCInstancetype_read(ExecState* execState, const void* buffer, J
     id value = *static_cast<const id*>(buffer);
     JSC::Identifier instancesStructureIdentifier = jsCast<GlobalObject*>(execState->lexicalGlobalObject())->instanceStructureIdentifier();
 
-    if (Structure* structure = jsDynamicCast<Structure*>(execState->thisValue().get(execState, instancesStructureIdentifier))) {
-        return toValue(execState, value, ^{
-            return structure;
-        });
+    Structure* structure;
+
+    if (ObjCConstructorBase* constructor = jsDynamicCast<ObjCConstructorBase*>(execState->thisValue())) {
+        structure = constructor->instancesStructure();
+    } else if (ObjCWrapperObject* wrapperObject = jsDynamicCast<ObjCWrapperObject*>(execState->thisValue())) {
+        structure = wrapperObject->structure();
+    } else if (ObjCSuperObject* superObject = jsDynamicCast<ObjCSuperObject*>(execState->thisValue())) {
+        structure = superObject->wrapperObject()->structure();
+    } else if (AllocatedPlaceholder* allocatedPlaceholder = jsDynamicCast<AllocatedPlaceholder*>(execState->thisValue())) {
```

Comment:
`AllocatedPlaceholder` is going to come up very often here. Please move this condition after the one for `ObjCConstructorBase`.